### PR TITLE
fix(operation): parse YAML frontmatter properly when rendering OPERATION.md

### PR DESCRIPTION
**Pull request by LangSensei**
## What
Use a real YAML parser for OPERATION.md so the dashboard no longer renders the leading frontmatter block as part of the Markdown body, and never leaks YAML block-scalar markers (`|`, `>`) into the `summary` field.

## Why
- Opening an operation detail page rendered the entire `---`-bounded YAML metadata block verbatim through marked.js (the `---` lines became horizontal rules, the keys became literal text).
- The `summary` field surfaced as raw YAML — for ops using `summary: |` or `summary: >`, the literal/folded markers leaked into the UI (see op `20260428-cde46d8c`). Root cause: extraction was not going through a real YAML parser.

todo id: `skip-frontmatter`

## Changes
- `frontmatter.go` (new):
  - `splitFrontmatter` — only strips when the file starts with `---` on line 1 followed by a real EOL, ending at the next standalone `---` line. Never touches body horizontal rules. Leaves unterminated frontmatter completely untouched.
  - `parseFrontmatter` — unmarshals the YAML block via `gopkg.in/yaml.v3` so block scalars resolve to plain Go strings.
  - `frontmatterString` — only returns string-typed values (numbers/maps/etc. → "") so type fallbacks cannot leak markers.
- `main.go`:
  - `handleOpFile` strips the frontmatter from `/api/file?file=OPERATION.md` responses, so the frontend's `marked.parse` only sees the body.
  - `opToView` re-reads `OPERATION.md` and overrides `Summary`/`Brief` from the parsed YAML map, falling back to the upstream-provided values on any error.
- `frontmatter_test.go` (new): all 7 edge cases from the brief plus CRLF, non-delimiter prefix, empty frontmatter, and non-string field handling.
- `go.mod` / `go.sum`: add `gopkg.in/yaml.v3 v3.0.1`.

## How to Test
```bash
go test ./...
```
All `TestSplitFrontmatter_*` / `TestParseFrontmatter_*` / `TestFrontmatterString_*` tests cover:
- (a) no frontmatter
- (b) frontmatter present
- (c) body contains `---` HR
- (d) frontmatter only, no body
- (e) unterminated frontmatter (must NOT be stripped)
- (f) `summary: |` literal block scalar
- (g) `summary: >` folded block scalar

Manual: open any operation in the dashboard whose OPERATION.md uses `summary: |` (e.g. `20260428-cde46d8c`). Summary should render as plain text without the `|`, and the rendered OPERATION.md tab should start at the body title, not at `---`.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -56,6 +56,22 @@ func opToView(op *operation.Operation) OpView {
 		Brief:   op.Brief,
 		Summary: op.Summary,
 	}
+	// Re-parse OPERATION.md frontmatter with a real YAML parser so block-
+	// scalar markers (`|`, `>`) and other YAML literals never leak into the
+	// UI. Falls back to the upstream-provided value on any error.
+	if home, err := os.UserHomeDir(); err == nil {
+		opMD := filepath.Join(home, ".swat", "squads", op.Squad, "operations", op.OperationID, "OPERATION.md")
```
Comment (LangSensei) [inline main.go]: [suggestion] This duplicates the path-building convention that already lives in `opDir` (main.go:292-298). Consider reusing it:

```go
if dir, err := opDir(op.OperationID); err == nil {
    opMD := filepath.Join(dir, "OPERATION.md")
    ...
}
```

That way the `~/.swat/squads/<squad>/operations/<id>` convention has a single source of truth and any future change (e.g. a configurable swat home) only needs to touch one place. Note: `opDir` itself calls `operation.Find` again, so a tiny helper that takes `op.Squad`+`op.OperationID` directly would avoid the double lookup and be even cleaner.

File: main.go [hunk 2]
```diff
@@ -56,6 +56,22 @@ func opToView(op *operation.Operation) OpView {
...
+			if meta, _, ok := parseFrontmatter(data); ok {
+				if s := frontmatterString(meta, "summary"); s != "" {
+					v.Summary = s
+				}
+				if s := frontmatterString(meta, "brief"); s != "" {
+					v.Brief = s
+				}
+			}
+		}
+	}
```
Comment (LangSensei) [inline main.go:77]: [suggestion] `opToView` is invoked for every operation returned by `scanOperations`, so this adds an `os.ReadFile` + YAML parse per operation on every list call (the `/api/ops` endpoint becomes O(N) disk reads). The brief explicitly flagged this concern. Two cheap mitigations:

1. Short-circuit when the upstream values already look clean — e.g. skip the re-read if `op.Summary` doesn't start with `|`/`>` and contains no leading newline.
2. Or push the YAML parse into the upstream `commander/operation` package once, so `op.Summary`/`op.Brief` arrive correctly here and the view layer doesn't need to know about frontmatter at all.

Not blocking at current operation counts, but worth a follow-up before the directory grows.

**Review by LangSensei — COMMENTED** [review]
Solid server-side fix. The `frontmatter.go` module is well-documented, leading-byte gating correctly prevents body horizontal rules from being treated as delimiters, CRLF and unterminated frontmatter are handled, and `frontmatterString` defensively ignores non-string values so block-scalar markers cannot leak. Tests cover all 7 brief edge cases plus CRLF, non-delimiter prefix, empty frontmatter, and non-string fields -- `go test ./...` passes locally.

Merge state: branch was created off `d69981f` (before PR #18 merged), so the GitHub UI may show a `scanOperations` diff that is not actually changed by this commit; `gh pr view --json mergeable` confirms `MERGEABLE` and `git show HEAD --stat` confirms the PR commit only adds 23 lines to `main.go`.

Note on PR #20 layering: with `handleOpFile` now stripping frontmatter server-side, `static/frontmatter.js` (added in PR #20, not in this branch) becomes a no-op on the OPERATION.md path because the response no longer starts with `---`. That's harmless defense-in-depth, no action needed -- just worth being aware of after merge.

Approving in spirit (event=COMMENT only because GitHub blocks self-APPROVE on this repo). Two non-blocking suggestions inline.

**Comment by LangSensei** [comment]
Pushed 0a7b1f8 to address the two `[suggestion]` items from review #4188786585.

**1. `opToView` duplicated path-building logic (review item 1)**
Extracted `opDirForSquad(squad, opID)` as the single source of truth for the `~/.swat/squads/<squad>/operations/<id>` convention. `opDir` keeps its `operation.Find` lookup and now delegates to it; `opToView` calls the pure helper directly with the squad it already has, so any future `swat home` refactor only needs to touch one place. (Calling `opDir` from inside the list loop would have created a *new* N+1 from `operation.Find`, so the helper is a deliberate split.)

**2. N+1 file reads on every list call (review item 2)**
Took the reviewer's option (a): added `needsFrontmatterReparse` and skip the `ReadFile` + YAML parse when neither `op.Brief` nor `op.Summary` starts with a YAML block-scalar marker (`|` or `>`, with optional leading whitespace and chomping/indent indicators). The common (already-clean) case is now zero-cost; only payloads that actually leak a marker pay for the re-parse.

**3. `static/frontmatter.js` (review item 3, info-only)**
Left in place per reviewer guidance  `handleOpFile` already strips frontmatter server-side, but the client helper is harmless defense-in-depth.

**Tests**
`go test ./...`  all green. Existing 11 frontmatter tests untouched; added `TestNeedsFrontmatterReparse` with 11 sub-cases covering: plain text, plain with newline, `|`/`>` markers, `|-` / `>+` chomping variants, leading space/tab before marker, only-whitespace input, and marker-not-at-start (should not trigger).

**Comment by LangSensei** [comment]
Closing  superseded by upstream fix [LangSensei/swat#95](https://github.com/LangSensei/swat/pull/95), which migrates `commander/operation` to `gopkg.in/yaml.v3`. Once dashboard bumps its `github.com/LangSensei/swat` dep past that commit, `Summary`/`Brief` arrive clean from upstream and the `opToView` re-read here becomes dead code. Frontend frontmatter stripping is already handled by PR #20, so the server-side `handleOpFile` strip is also redundant.

Will track the dep bump as a separate, much smaller change. Cancelling the in-flight review-fix op `20260428-8b9cd174` as well.
